### ISSUE TITLE
Does anybody want to develop a ruleset for umatrix?

### ISSUE THREAD

**Issue by AndrewtheCanon** [drduh/macOS-Security-and-Privacy-Guide#396]
I am currently developing a rule-set for umatrix, given all the thirdparty tracking sites. Does anybody want to contribute to it?

We can do this in this repository (if drduh is ok with it) or outside

I didn't find too many good rule-sets for umatrix

Do the blacklists already block these tracking sites?

**Comment by AndrewtheCanon** [comment]
Has anybody verified the blocklists mentioned in the main article are blocking the trackers on most of the websites?

umatrix shows a huge number of trackers

**Comment by TraderStf** [comment]
umatrix is no more maintained (status: public archive), must use ublockorigin

https://www.ghacks.net/2021/07/15/umatrix-has-an-unfixed-vulnerability-here-is-a-workaround/

**Comment by beerisgood** [comment]
Badness enumeration don’t work. 
you will end in endless maintenance and a lot broken sites. 

This can also be done with DNS and lists like OISD in a lot better way.

**Comment by AndrewtheCanon** [comment]
Yes, for sure, you can do it one at a time and also specifically block loading from sources on per site basis. The advantage of uMatrix is that it shows all the sites that are loading right on the interface, which makes it much more convenient to block sites that are not relevant.

Do you have a way to do that using uBlockOrigin, WireShark etc, as quickly?

I have seen that certain resources that are necessary for certain websites to render correctly, are not necessary for others. uMatrix allows you to do per site blocking, which speeds up loading sites a lot.

Does uBlockOrigin/OISD allow you to do that, as quickly?

**Comment by AndrewtheCanon** [comment]
> Badness enumeration don’t work.
> you will end in endless maintenance and a lot broken sites.

Its not that bad. I have had a fixed configuration for a while for sites that are well-tested, and it works very well repeatedly. Ofcourse overtime, certain sites could add other sources, which would requires some maintenance

**Comment by johnsoga** [comment]
This repo in general I believe manages to stay very relevant by staying up to date with this information it provides. While I do see blocklists as a useful tool and technology as pointed out about the [uMatrix](https://github.com/gorhill/uMatrix) repo has been archived. Instructing use of outdated tools in the security leads to all kinds of problems. If there is an active tool being used that has these same features I think that could be useful but otherwise I would leave uMatrix out

**Comment by AndrewtheCanon** [comment]
Thats a good point @johnsoga. uBlockOrigin is mostly an adblocker. Do you know anything else like uMatrix? Which is more like a web firewall with granular permissions?

**Comment by johnsoga** [comment]
@AndrewtheCanon so admittedly I cannot say I was well versed with the uMatrix utility so I took a bit of time to take a look at the repo. Looking it over I have to say I really don't see the difference compared to UBlockOrigin. I kind of disagree with this idea of calling it a "web firewall" but maybe I'm just missing what this particular functionality is that it has to be called such.



> I am currently developing a rule-set for umatrix, given all the thirdparty tracking sites. Does anybody want to contribute to it?

Sounds like your trying to build a block list for ads/trackers which is what uBlockOrigin allows. Which I'd be more than happy to help contribute towards. Let me know how I can help.

> Yes, for sure, you can do it one at a time and also specifically block loading from sources on per site basis. The advantage of uMatrix is that it shows all the sites that are loading right on the interface, which makes it much more convenient to block sites that are not relevant.
>
> Do you have a way to do that using uBlockOrigin, WireShark etc, as quickly?

Clicking on the UBlockOrigin plugin icon it looks there is a "Open the Logger" option that seemed to do what you were asking for when I tried it. It showed all the elements/domains being loaded on the web page I was on. I think this might be what you are looking for. I don't think wireshark would be necessary here and would probably be a lot harder to filter through to get the result you are looking for.

> I have seen that certain resources that are necessary for certain websites to render correctly, are not necessary for others. uMatrix allows you to do per site blocking, which speeds up loading sites a lot.
> 
> Does uBlockOrigin/OISD allow you to do that, as quickly?

UBlockOrigin seems to support [Dynamic Filtering](https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-quick-guide) which I think supports what you are asking for using local dynamic filtering rules
